src/cmd2.cc

```diff
@@ -556,58 +545,9 @@ void do_cmd_go_down(void)
 
 		/* Leaving */
 		p_ptr->leaving = TRUE;
-
-		if (!fall_trap)
-		{
```

Comment:
Yeah, staircases work fine here. I *think* the logic was just confused due to the special trap case.